ragaai_catalyst/ragaai_catalyst.py

```
@@ -143,78 +149,133 @@
-            logger.error(
-                "Token retrieval was not successful: %s",
-                token_response.get("message", "Unknown error"),
-            )
-            return None
+        # Case 1: No token - synchronous retrieval (fail fast)
+        if not current_token:
+            return self.get_token(force_refresh=True)
+
+        # Case 2: Token expired or about to expire - async refresh
+        if not self._token_expiry or current_time >= self._token_expiry:
+            if not self._refresh_thread or not self._refresh_thread.is_alive():
```

Ruff (0.8.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)